### ISSUE TITLE
`with` scope-function's side effect tracking.

### ISSUE THREAD

**Issue by stefanusayudha** [pinterest/ktlint#2474]
## Expected Rule behavior
- when using `with` scope - side effect that targeting the context should be called (suggested) in `apply` block.
  ```kotlin
  // wrong example
  with(CarObj) {
     color = getNewColor(..)
  }
  ```
  ```kotlin
  // correct example
  with(CarObj) {
     val newColor = getNewColor(..)
     
     apply {
       color = newColor
     }
     
  }
  ```
- `apply` block inside the `with` scope should be only 1 (suggested not more than 1).
- when using `with` scope - side effect that targeting something outside the function should be called in `also` scope.
  ```kotlin
  
  var currentCarColor = SOME_COLOR
  
  with(CarObj) {
     val newColor = getNewColor(..)
     
     apply {
       color = newColor
     }
     
     also {
       currentCarColor = newColor
     }
  }
  ```
- `also` scope inside `with` scope should be only 1 (suggested not more than 1).

Those rule, improve the understanding of what side effects is happening in the code for new developer. Also forcing the developer to pay attentions to the side effects. 

Since i can't find any source that suggesting this method, i cannot give any references for this. Since this feature is not fundamental of how the language is working, i don't recommend this to become a basic feature (activated by default), but as an optional feature. And i believe people gonna love this, espescialy I am. Big thanks.

**Comment by stefanusayudha** [comment]
i think this feature is too much 😅

**Comment by paul-dingemans** [comment]
Tnx for opening *and* closing this issue. This would indeed be to far out of scope for ktlint.
